ops/testing.py

```diff
@@ -3023,6 +3039,8 @@ def __init__(self, backend: _TestingModelBackend, container_root: pathlib.Path):
         self._exec_handlers: Dict[Tuple[str, ...], ExecHandler] = {}
         self._notices: Dict[Tuple[str, str], pebble.Notice] = {}
         self._last_notice_id = 0
+        self._changes: Dict[str, pebble.Change] = {}
+        self._check_infos: Dict[str, pebble.CheckInfo] = {}
```

Comment:
"CheckInfo" is a thing in pebble.
At the same time, `self._check_infos` just reads unclear/muddy.
I wish we could name things a bit better, but I guess in this case we have to keep pebble (or juju) nomenclature, don't we?
cc @benhoyt

Reply:
Do you have a suggestion for a better name? I don't love `_check_infos` either, but it's accurate and consistent, and since it's private I don't care as much.

Reply:
I first wrote a long justification for Tony's way (summary at the end), but since the ask is specifically for a suggestion or an alternative here it goes... please tread lightly:

### Alternative

Revisit https://github.com/canonical/operator/pull/668

Rename `CheckInfo` (which mirror go internal structure and a JSON block in the API response) to something like `CheckStatus` or `DetailedCheckStatus`, because that's what (I think) it is.

Now is probably the last opportunity to change names, given that this PR makes responding to check failures functional.

Caveats:
We'd have to make a call on offering an alias for (hypothetical?) existing users of the public `ops.pebble.*` API.

CheckInfoMapping is re-exported at top level, but CheckInfo is not; it seems that only jhack uses this type directly; Only sunbeam seems to use `.get_checks()`.

### Keep Current

The library's job is to expose specific juju and pebble concepts, so I guess we have to stay close to the original naming.

This particular member is "private", thus it's not a burden to our users, and it's annotated with a more readable type (I imagine the key is the check name that's defined in the pebble plan; and the value is a typed thing).

Reply:
We discussed this briefly in our daily sync. I think you're right that `CheckStatus` is a better name, but we have 4 "FooStatus" types in pebble.py (and have for a long time), so if we changed one, it'd be inconsistent with the others. Renames like this can cost quite a lot (cf "integration" vs "relation"), so I think we should just stick with the status quo here. `CheckInfo` is not sooo bad.